### ISSUE TITLE
Add Airflow TaskGroup run information

### ISSUE THREAD

**Issue by collado-mike** [OpenLineage/OpenLineage#987]
Airflow tasks can belong to a [Task Group](https://airflow.apache.org/docs/apache-airflow/stable/concepts/dags.html#taskgroups), which organizes tasks, typically by prepending the name of the group to the task name, separated by a dot (.) - e.g., `my_task_group.my_task`. OpenLineage typically just reports the full name of the task as the job name (e.g., `my_dag.my_task_group.my_task`), but the name of the DAG is reported in the `ParentRunFacet`. We should also report the `TaskGroup` information so that OpenLineage consumers can know which jobs belong to various groups. Ideally, each `TaskGroup` can be reported as a new parent job run, with a distinct `RunId` so that consumers can compare past runs with new ones. However, this might be difficult, as there would also need to be a Run event that tied the `TaskGroup` run to the parent `DagRun` and `TaskGroup`s don't actually have a start or complete event. It's possible to simply report a running event for a `TaskGroup` every time a task runs so that each task run would report two events - one for the `TaskGroup` run and one for the task itself. But `TaskGroup`s can also be nested, so reporting the full hierarchy would become tricky. It's likely that simply reporting the `TaskGroup` information as a facet would allow OpenLineage consumers to do the untangling server side, rather than trying to fake events in Airflow itself.

**Comment by mobuchowski** [comment]
AFAIK `TaskGroups` are just UI grouping concept. They don't affect the actual execution, which still is tied to particular DAG run. Maybe we should add "generic" grouping facet that consumers like `Marquez` could recognize and "fold" the resulting runs belonging to one group. 

This could also be used in a Spark integration where one SQL statement spawns several Spark jobs, like Delta Lake operations.

**Comment by julienledem** [comment]
I agree with Maciej that this should be captured with a generic grouping facet and not with a Run. 
Let's reword this ticket accordingly.

**Comment by mobuchowski** [comment]
@JDarDagran fyi
